deploy

```diff
@@ -154,13 +154,19 @@ def process_template(args):
         print(f"  {k}={v}")
     print()
 
-    def gen_param_args():
-        return [q for k, v in params.items() for q in ['--param', f'{k}={v}']]
+    def gen_param_args(selected):
+        selected_params = {(k, v) for k, v in params.items() if k in selected}
+        return [q for k, v in selected_params for q in ['--param', f'{k}={v}']]
```

Comment:
I think the part that was confusing to me here was the double for loop inside a list comprehension.. here's my alternative that I can read better:

```
def gen_param_args(selected):                                             
    selected_params = {(k, v) for k, v in params.items() if k in selected}
    args = []                                                             
    for k, v in selected_params:                                          
        args += ['--param', f'{k}={v}']                                   
    return args                                                           
                                                                          

```

I'll understand if you don't want to make this change.

Reply:
Not sure where this stands on the idiomatic scale (I've definitely seen it used elsewhere), but it's really useful. Here's how to read this: when you see something like `[w for y in z for x in y for w in x]`, you can simply unroll it in that same order:

```python
for y in z:
  for x in y:
    for w in x:
      mylist.append(w)
```